_assets/scripts/migration_check.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+set -e
+set -o pipefail
+
+check_migration_order() {
+  local prev_migration=""
+  for file in "$@"; do
+    echo "this ${file}"
+    current_migration=$(echo "$file" | cut -d'-' -f1)
+
+    if [[ ! -z "$prev_migration" && "$current_migration" < "$prev_migration" ]]; then
+      echo "Error: Migration files are out of order. Please ensure migrations are added in chronological order."
+      exit 1
+    fi
+
+    prev_migration="$current_migration"
+  done
+}
+
+committed_files=$(git ls-tree -r --name-only HEAD protocol/migrations/sqlite/*.sql | sort)
+staged_files=$(git diff --name-only develop protocol/migrations/sqlite/*.sql | sort)
```

Comment:
I see this is failing with:
```
fatal: ambiguous argument 'develop': unknown revision or path not in the working tree.
```
Maybe we could use `origin/develop` here:
```suggestion
staged_files=$(git diff --name-only origin/develop protocol/migrations/sqlite/*.sql | sort)
```